lib/Sema/IDETypeCheckingRequests.cpp

```diff
@@ -156,6 +156,12 @@ static bool isMemberDeclAppliedInternal(const DeclContext *DC, Type BaseTy,
   GenericSignature genericSig = genericDecl->getGenericSignature();
   if (!genericSig)
     return true;
+  if (isa<TypeAliasDecl>(VD) && BaseTy->is<ProtocolType>()) {
```

Comment:
Could we do this before line 153, before getting the generic signature?

Reply:
Good catch, makes sense as it’s cheaper to check. Moved it up